client/app/pages/dashboards/dashboard.js

```diff
@@ -26,6 +27,32 @@ function DashboardCtrl($rootScope, $routeParams, $location, $timeout, $q, $uibMo
     }
   };
 
+  const extractGlobalParameters = () => {
+    let globalParams = {};
+    this.dashboard.widgets.forEach(row =>
+      row.forEach((widget) => {
+        widget.getQuery().getGlobalParametersDefs().forEach((param) => {
+          const defaults = {};
+          defaults[param.name] = _.clone(param);
+          defaults[param.name].locals = [];
+          globalParams = _.defaults(globalParams, defaults);
+          globalParams[param.name].locals.push(param);
+        });
+      })
+    );
+    this.globalParameters = _.values(globalParams);
+  };
+
+  this.onGlobalParametersChange = () => {
+    _.each(this.globalParameters, (global) => {
+      _.each(global.locals, (local) => {
+        local.value = global.value;
+      });
+    });
+  };
+
+  $scope.$on('deleteDashboardWidget', extractGlobalParameters);
```

Comment:
Instead of using an event, let's add a callback function to the widget component.